Add disable auto assign flag

msoderberg opened the pull request:
**What this PR does / why we need it**:
A proposed implementation of the "disable auto assign" feature brought up in #4 

**Which issue(s) this PR fixes**:
Fixes #4 

**Special notes for your reviewer**:
Defaulting the flag to true required some changes in the test configurations. Maybe it would be cleaner to name it `disable-auto-assign` so we can go with the zero bool value if the flag has not been set in the YAML?

danderson commented on internal/config/config.go:
```diff
@@ -82,6 +83,8 @@ type Pool struct {
 	// unusable, for maximum compatibility with ancient parts of the
 	// internet.
 	AvoidBuggyIPs bool
+	// Prevents IP addresses to be automatically assigned from this pool.
```
Can you start the docstring with "if false, ..." here? Otherwise this docstring sounds like I have to set this = true to prevent auto-allocation.

danderson commented on manifests/example-config.yaml:
```diff
@@ -43,6 +43,10 @@ data:
       # smurf protection. Such devices have become fairly rare, but
       # the option is here if you encounter serving issues.
       avoid-buggy-ips: true
+      # (optional) If false, MetalLB will not automatically allocate
```
Can you say `(optional, default true)`, to make it clear that you have to disable it?

msoderberg commented:
Thanks for the feedback! I've updated accordingly. I added a section in the usage docs but I basically copied your description from #4. Feel free to edit as you wish before you merge :)

danderson commented:
Thank you for contributing! Merged.
